Packs/CommonScripts/Scripts/GenericPollingScheduledTask/GenericPollingScheduledTask.js

```diff
@@ -65,71 +80,167 @@ function setNextRun(ids, playbookId, pollingCommand, pollingCommandArgName, pend
-        pendingPath = playbookContext + "." + args.pendingIds;
-    }
-    var pendings = dq(invContext, pendingPath);
+        if ('playbookId' in args) {
+            playbookContext = 'subplaybook-' + args.playbookId;
+            pendingPath = playbookContext + "." + args.pendingIds;
+        }
+        var pendings = dq(invContext, pendingPath);
 
-    if (pendings === null) {
-        return finish(args.playbookId, args.tag);
-    }
+        if (pendings === null) {
+            return finish(args.playbookId, args.tag);
+        }
+
+        var idsStrArr = listOfStrings(ids);
+        var pendingsStrArr = listOfStrings(pendings);
+        idsToPoll = intersect(idsStrArr, pendingsStrArr);
+        if (idsToPoll.length === 0) {
+            return finish(args.playbookId, args.tag);
+        }
+
+        // Run the polling command for each id
+        var pollingCommandArgs = {};
+        var names = argToList(args.additionalPollingCommandArgNames);
+        var values = argToList(args.additionalPollingCommandArgValues);
+
+        for (var index = 0; index < names.length; index++)
+            pollingCommandArgs[names[index]] = values[index];
+
+        pollingCommandArgs[args.pollingCommandArgName] = idsToPoll.join(',');
+        var res = executeCommand(args.pollingCommand, pollingCommandArgs);
+
+        // Change the context output of the polling results to the local playbook context
+        if ('playbookId' in args) {
+            for (var i = 0; i < res.length; i++) {
+                if ('EntryContext' in res[i]) {
+                    for (var k in res[i].EntryContext) {
+                        res[i].EntryContext[playbookContext + "." + k] = res[i].EntryContext[k];
+                        delete res[i].EntryContext[k];
+                    }
+                }
+            }
+        }
+
+        // Schedule the next iteration
+        var scheduleTaskRes = setNextRun(args.ids, args.playbookId, args.pollingCommand, args.pollingCommandArgName, args.pendingIds, args.interval, args.timeout, args.tag, args.additionalPollingCommandArgNames, args.additionalPollingCommandArgValues);
+        if (isError(scheduleTaskRes[0])) {
+            res.push(scheduleTaskRes);
+        }
 
-    var idsStrArr = listOfStrings(ids);
-    var pendingsStrArr = listOfStrings(pendings);
-    idsToPoll = intersect(idsStrArr, pendingsStrArr);
-    if (idsToPoll.length === 0) {
-        return finish(args.playbookId, args.tag);
+        return res;
+    }
+    catch (err) {
+        finish(args.playbookId, args.tag, err);
+        throw err;
     }
+}
 
-    // Run the polling command for each id
-    var pollingCommandArgs = {};
-    var names = argToList(args.additionalPollingCommandArgNames);
-    var values = argToList(args.additionalPollingCommandArgValues);
+function genericPollingScheduledTaskWithGuid() {
+    try {
+        var guid = args.scheduledEntryGuid;
+        var endTime = stringToDate(args.endTime, "%Y-%m-%d %H:%M:%S");
```

Comment:
Don't we need to add endTime as a script arg here?

Reply:
We did ([link](https://github.com/demisto/content/pull/26372/files#diff-e28697ad2539513e99870ee66b6ce01287e116854ed5f92eb601e00174dc4272:~:text=name%3A%20endTime,end%20the%20polling.))